src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -177,6 +178,110 @@ internal static unsafe HttpMethod GetKnownMethod(byte* data, int length, out int
             return HttpMethod.Custom;
         }
 
+        /// <summary>
+        /// Parses string <paramref name="value"/> for a known HTTP method.
+        /// </summary>
+        /// <remarks>
+        /// A "known HTTP method" can be an HTTP method name defined in the HTTP/1.1 RFC.
+        /// The Known Methods (CONNECT, DELETE, GET, HEAD, PATCH, POST, PUT, OPTIONS, TRACE)
+        /// </remarks>
+        /// <returns><c>true</c> if the input matches a known string, <c>false</c> otherwise.</returns>
+        public static bool GetKnownMethod(string value, out HttpMethod method)
```

Comment:
Probably should be called `TryGetKnownMethod` since it is following the try pattern of returning a bool with an out param.

Reply:
Changed

Reply:
Reverted, always returns value in `out`

Reply:
What's the bool for then?

Reply:
Ah good point can clean this up; return the string